src/libraries/System.IO.Ports/src/System/IO/Ports/SerialPort.Unix.cs

```diff
@@ -5,13 +5,28 @@
 using System.ComponentModel;
 using System.Collections.Generic;
 using System.IO;
+using System.Runtime.InteropServices;
 
 namespace System.IO.Ports
 {
     public partial class SerialPort : Component
     {
         public static string[] GetPortNames()
         {
+#if NETCOREAPP
+            return OperatingSystem.IsLinux() ? GetPortNames_Linux()
+                : OperatingSystem.IsMacOS() ? GetPortNames_OSX()
+                : OperatingSystem.IsFreeBSD() ? GetPortNames_FreeBSD()
+#else
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? GetPortNames_Linux()
+                : RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? GetPortNames_OSX()
+                : RuntimeInformation.IsOSPlatform(OSPlatform.Create("FREEBSD")) ? GetPortNames_FreeBSD()
```

Comment:
should you create a variable for FreeBSD?

Reply:
Why would I need a variable? The OSPlatform checks happens only once.

Reply:
The order looks right. The `IsOSPlatform` will be called once for Linux, twice for OSX and three times for FreeBSD. That make sense as well as the comparisons is very cheap AFAIK.